### PR TITLE
Add Parental responsibility order (C45A)

### PR DESCRIPTION
**Pull request by jacmurphy**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/FPLA-832

### Change description ###

As a Judge or HMCTS Admin
I want to be able to create Section 4 orders in the service
So that I can order who has parental responsibility for the children

Acceptance Criteria

    In Manage orders, add new option to list of orders in create an order flow (alphabetically):
        Parental responsibility order (C45A)
    Additional order details page:
        Is the order by consent?
        Applicant name (this information could be played back as should be got from the application link)
        Further directions 
        Is the order final?
            If so, ask the close case question
    Order template contains Notice content
    On completion of the event:
        Store pre-sealed Word version of the order (for slip rule amendments later)
        Convert to pdf
        Seal the order
        Display order in the Orders tab (most recent order at the top)
    Tab display details:
        Order type (Parental responsibility order)
        Order start date
        Name of applicant with parental responsibility
        Name of child/children if not all children on the case included
    Send order issued notifications to parties according to serving preference
    Toggle off for release

 Example of output
Screen Shot 2021-07-15 at 14 55 23

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: ccd-definition/FixedLists/CareSupervision/ManageOrdersType.json [hunk 1]
```diff
@@ -62,18 +62,25 @@
     "ListElement": "Interim supervision order (C35B)",
     "DisplayOrder": 9
   },
+  {
+    "LiveFrom": "01/01/2017",
+    "ID": "ManageOrdersType",
+    "ListElementCode": "C45A_PARENTAL_RESPONSIBILITY_ORDER",
```
Comment (samuelzerozero) [inline ccd-definition/FixedLists/CareSupervision/ManageOrdersType.json:68]: If you are moving it here, you want to delete it from ManageOrdersUploadType

File: service/src/main/java/uk/gov/hmcts/reform/fpl/model/order/Order.java [hunk 2]
```diff
@@ -313,12 +314,22 @@
...
+        List.of(
+            LINKED_TO_HEARING,
+            LINK_APPLICATION,
+            APPROVER,
+            APPROVAL_DATE_TIME,
+            WHICH_CHILDREN,
+            ORDER_BY_CONSENT,
+            PARENTAL_RESPONSIBILITY,
+            FURTHER_DIRECTIONS,
+            REVIEW_DRAFT_ORDER
```
Comment (samuelzerozero) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/model/order/Order.java:331]: Is WHICH_OTHERS not needed?

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/orders/prepopulator/question/ParentalResponsibilityPrePopulator.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+        DynamicList linkedApplication = caseData.getManageOrdersEventData().getManageOrdersLinkedApplication();
+
+        UUID selectedApplicationId = linkedApplication.getValueCodeAsUUID();
+        ApplicationsBundle selectedApplicationBundle = caseData.getApplicationBundleByUUID(selectedApplicationId);
+
+        if (selectedApplicationBundle instanceof C2DocumentBundle) {
+            C2DocumentBundle c2DocumentBundle = (C2DocumentBundle) selectedApplicationBundle;
+
+            if (c2DocumentBundle.getC2AdditionalOrdersRequested().contains(PARENTAL_RESPONSIBILITY)) {
```
Comment (samuelzerozero) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/service/orders/prepopulator/question/ParentalResponsibilityPrePopulator.java:55]: for readibility I'd return a map here. That will reduce the overload of having to go through the possible decision points

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/orders/prepopulator/question/ParentalResponsibilityPrePopulator.java [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+                data.put(PARENT_RESPONSIBLE,
+                    StringUtils.substringBefore(c2DocumentBundle.getApplicantName(), SEPARATOR));
+                mapParentResponsibility(data, c2DocumentBundle.getParentalResponsibilityType());
+            }
+
+        } else {
+            OtherApplicationsBundle otherApplicationsBundle = (OtherApplicationsBundle) selectedApplicationBundle;
+            if (otherApplicationsBundle.getApplicationType() == C1_PARENTAL_RESPONSIBILITY) {
```
Comment (samuelzerozero) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/service/orders/prepopulator/question/ParentalResponsibilityPrePopulator.java:63]: for readibility I'd return a map here. That will reduce the overload of having to go through the possible decision points

**Review by samuelzerozero — COMMENTED** [review]
no changes in the template?

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/orders/prepopulator/question/ParentalResponsibilityPrePopulator.java (hunk 4 above)
Comment (jacmurphy) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/service/orders/prepopulator/question/ParentalResponsibilityPrePopulator.java]: updated

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/orders/prepopulator/question/ParentalResponsibilityPrePopulator.java (hunk 3 above)
Comment (jacmurphy) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/service/orders/prepopulator/question/ParentalResponsibilityPrePopulator.java]: update on call with Sam

File: service/src/main/java/uk/gov/hmcts/reform/fpl/model/order/Order.java (hunk 2 above)
Comment (jacmurphy) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/model/order/Order.java:332]: Added

File: ccd-definition/FixedLists/CareSupervision/ManageOrdersType.json (hunk 1 above)
Comment (jacmurphy) [reply ccd-definition/FixedLists/CareSupervision/ManageOrdersType.json:68]: moved

File: service/build.gradle [hunk 5]
```diff
@@ -272,6 +272,7 @@ def versions = [
 ]
 
 dependencies {
+  implementation 'junit:junit:4.13.1'
```
Comment (samuelzerozero) [inline service/build.gradle]: ?

File: service/src/test/java/uk/gov/hmcts/reform/fpl/model/order/OrderTest.java [hunk 6]
```diff
@@ -214,6 +219,10 @@ void checkCoverage() {
             Arguments.of(C43A_SPECIAL_GUARDIANSHIP_ORDER, ORDER_DETAILS, Optional.of(REVIEW)),
             Arguments.of(C43A_SPECIAL_GUARDIANSHIP_ORDER, REVIEW, Optional.of(OTHER_DETAILS)),
             Arguments.of(C43A_SPECIAL_GUARDIANSHIP_ORDER, OTHER_DETAILS, Optional.empty()),
+            Arguments.of(C45A_PARENTAL_RESPONSIBILITY_ORDER, ISSUING_DETAILS, Optional.of(CHILDREN_DETAILS)),
+            Arguments.of(C45A_PARENTAL_RESPONSIBILITY_ORDER, CHILDREN_DETAILS, Optional.of(ORDER_DETAILS)),
+            Arguments.of(C45A_PARENTAL_RESPONSIBILITY_ORDER, ORDER_DETAILS, Optional.of(REVIEW)),
+            Arguments.of(C45A_PARENTAL_RESPONSIBILITY_ORDER, REVIEW, Optional.of(OTHER_DETAILS)),
```
Comment (samuelzerozero) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/model/order/OrderTest.java:225]: missing the last transition from OTHER DETAILS -> emtpy()

**Review by samuelzerozero — DISMISSED** [review]

**Review by samuelzerozero — DISMISSED** [review]

File: service/src/test/java/uk/gov/hmcts/reform/fpl/model/order/OrderTest.java (hunk 6 above)
Comment (jacmurphy) [reply service/src/test/java/uk/gov/hmcts/reform/fpl/model/order/OrderTest.java:225]: updated

File: e2e/tests/adminManagesOrders_test.js [hunk 7]
```diff
@@ -486,6 +486,38 @@ Scenario('Create (C26) Secure accommodation order (deprivation of liberty)', asy
...
+  await I.goToNextPage();
+  manageOrdersEventPage.selectOrderByConsent();
+  await manageOrdersEventPage.enterNameOfParentResponsible('T.J. Detweiler');
+  await manageOrdersEventPage.selectFatherAsResponsible();
+  await manageOrdersEventPage.enterFurtherDirections('Further details. '.repeat(10));
+  await manageOrdersEventPage.selectIsFinalOrder();
+  await I.goToNextPage();
+  await manageOrdersEventPage.checkPreview();
+  await I.goToNextPage();
+  pause();
```
Comment (samuelzerozero) [inline e2e/tests/adminManagesOrders_test.js:510]: that's the reason why is timing out :D

**Review by samuelzerozero — DISMISSED** [review]

**Review by samuelzerozero — APPROVED** [review]

**Pull request merged**